### PR TITLE
feat: add dependabot to monitor dependencies

### PR DESCRIPTION
**Pull request by rdimitrov**
The following PR adds Dependabot to monitor for dependency updates

Fixes: https://github.com/vmware/repository-service-tuf/issues/56 and https://github.com/vmware/repository-service-tuf/issues/57

Signed-off-by: Radoslav Dimitrov <dimitrovr@vmware.com>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+updates:
+  # Monitor Github Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+      time: "10:00"
+    commit-message:
+      prefix: "chore"
+      include: "scope"
```
Comment (kairoaraujo) [inline .github/dependabot.yml:11]: IMO is better to use a specific `prefix` for GHA commit messages such as `"gha"`

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    open-pull-requests-limit: 10
+  # Monitor Python test dependencies
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
+      time: "10:00"
+    commit-message:
+      prefix: "chore"
+      include: "scope"
```
Comment (kairoaraujo) [inline .github/dependabot.yml:21]: Same here, we could have like `prefix`: `requirements`
I also suggest adding the `prefix-development`: `requirements dev`

**Review by kairoaraujo — CHANGES_REQUESTED** [review]
Thank you @rdimitrov !

File: .github/dependabot.yml (hunk 1 above)
Comment (rdimitrov) [reply .github/dependabot.yml:11]: We can do that, but I thought to not diverge from the general practice - ref. https://www.conventionalcommits.org/en/v1.0.0/#summary 

In that sense, the spec more or less combines everything dependency related under `build`. 

Do we prefer to have everything under `chore`(or `build`) or have separate ones like `gha`, `pip`, `pip dev`, and so forth?

Comment (kairoaraujo) [reply .github/dependabot.yml:11]: Got it!

File: .github/dependabot.yml (hunk 2 above)
Comment (rdimitrov) [reply .github/dependabot.yml]: Done 👍

File: .github/dependabot.yml (hunk 1 above)
Comment (rdimitrov) [reply .github/dependabot.yml]: Settled to `build-dev` for development related dependencies and `build` for everything else 😃

**Review by kairoaraujo — APPROVED** [review]

**Pull request merged**